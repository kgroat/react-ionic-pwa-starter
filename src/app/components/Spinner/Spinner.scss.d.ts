export const spinner: string;
export const rotator: string;
export const path: string;
export const dash: string;
