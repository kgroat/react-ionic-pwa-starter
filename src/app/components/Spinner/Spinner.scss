
$offset: 280;
$duration: 1.4s;

.spinner {
  animation: rotator $duration * 0.83 linear infinite;
}

@keyframes rotator {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.path {
  stroke-dasharray: $offset;
  stroke-dashoffset: 0;
  transform-origin: center;
  animation: dash $duration ease-in-out infinite;
  stroke: currentColor;
}

@keyframes dash {
 0% { stroke-dashoffset: $offset; }
 50% {
   stroke-dashoffset: $offset/4;
   transform:rotate(180deg);
 }
 100% {
   stroke-dashoffset: $offset;
   transform:rotate(360deg);
 }
}