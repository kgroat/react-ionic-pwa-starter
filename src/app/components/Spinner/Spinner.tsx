
import * as React from 'react'

import { spinner, path } from './Spinner.scss'

export interface Props {
  className?: string
  style?: React.CSSProperties
}

export default class Spinner extends React.Component<Props> {
  render () {
    const { className = '', style } = this.props

    return (
      <svg className={`${spinner} ${className}`} viewBox='0 0 100 100' xmlns='http://www.w3.org/2000/svg' style={style}>
        <circle className={path} fill='none' strokeWidth={10} strokeLinecap='round' cx={50} cy={50} r={45} />
      </svg>
    )
  }
}
