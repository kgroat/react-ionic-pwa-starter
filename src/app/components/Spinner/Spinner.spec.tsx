
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import Spinner from './Spinner'

describe(`<${Spinner.name} />`, () => {
  describe('shapshots', () => {
    it('should render', () => {
      const element = create((
        <Spinner />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given className', () => {
      const element = create((
        <Spinner className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given style', () => {
      const element = create((
        <Spinner style={{ top: 0, left: 0, right: 0, bottom: 0 }} />
      ))

      expect(element).toMatchSnapshot()
    })
  })
})
