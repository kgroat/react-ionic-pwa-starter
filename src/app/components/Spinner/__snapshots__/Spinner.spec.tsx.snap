// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> shapshots should render 1`] = `
<svg
  className="spinner "
  style={undefined}
  viewBox="0 0 100 100"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="path"
    cx={50}
    cy={50}
    fill="none"
    r={45}
    strokeLinecap="round"
    strokeWidth={10}
  />
</svg>
`;

exports[`<Spinner /> shapshots should render with the given className 1`] = `
<svg
  className="spinner myClass"
  style={undefined}
  viewBox="0 0 100 100"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="path"
    cx={50}
    cy={50}
    fill="none"
    r={45}
    strokeLinecap="round"
    strokeWidth={10}
  />
</svg>
`;

exports[`<Spinner /> shapshots should render with the given style 1`] = `
<svg
  className="spinner "
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  viewBox="0 0 100 100"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="path"
    cx={50}
    cy={50}
    fill="none"
    r={45}
    strokeLinecap="round"
    strokeWidth={10}
  />
</svg>
`;
